runtime/compiler/z/codegen/S390PrivateLinkage.cpp

```diff
@@ -1649,6 +1659,140 @@ J9::Z::PrivateLinkage::buildNoPatchingVirtualDispatchWithResolve(TR::Node *callN
+   ccInterfaceDataAddress->iTableIndex = 0;
+   ccInterfaceDataAddress->j2iThunkAddress = virtualThunk;
+   ccInterfaceDataAddress->totalSizeOfPicSlots = (3 /*Total Number of PICSlots*/ * sizeof(ccInterfaceDataAddress->slot1Class)) + (3 * sizeof(ccInterfaceDataAddress->slot1Method));
+   ccInterfaceDataAddress->isCacheFull = 0;
+
+   // Now we are checking each slot and dispatching methods. 
+   intptr_t interfaceDataAddress = reinterpret_cast<intptr_t>(ccInterfaceDataAddress);
+
+   TR::StaticSymbol *interfaceDataSymbol =
+      TR::StaticSymbol::createWithAddress(trHeapMemory(), TR::Address, reinterpret_cast<void *>(interfaceDataAddress));
+   interfaceDataSymbol->setNotDataAddress();
+   TR::SymbolReference *interfaceDataSymRef = new (trHeapMemory()) TR::SymbolReference(cg()->comp()->getSymRefTab(), interfaceDataSymbol, 0);
+
+   /**
+    * LARL  regEP, interfaceDataAddress
+    * LARL  regRA, returnAddress
+    * Label StartICF: (Attaching predependency conditions here.
+    * LPQ   Class,MethodAddress, 0(interfaceDataReg)
+    * CGR   Class,vftReg
+    * BCR   Equal,MethodAddress
+    * LPQ   Class,MethodAddress, 16(interfaceDataReg)
+    * CGR   Class,vftReg
+    * BCR   Equal,methodAddress
+    * LPQ   Class,MethodAddress, 32(interfaceDataReg)
+    * CGR   Class,vftReg
+    * BCR   Equal,methodAddress
+    * Label SnippetCal: (Cache miss, will be calling the snippet)
+    * BRCL  interfaceCallSnippet
+    * Label doneLabel: (Attaching Post Deps Here)
+    */
+   int32_t numberOfAdditionalPostDeps = 3; // For load pair instruction, we need 3 virtual registers, even reg, odd reg, even-odd pair
+   TR::Register *regRA = dependencies->searchPostConditionRegister(getReturnAddressRegister());
+   if (regRA == NULL)
+      {
+      regRA = cg()->allocateRegister();
+      numberOfAdditionalPostDeps+=1;
+      }
+   TR::Register *regEP = dependencies->searchPostConditionRegister(getEntryPointRegister());
+   if (regEP == NULL)
+      {
+      regEP = cg()->allocateRegister();
+      numberOfAdditionalPostDeps+=1;
+      }
+
+   TR::Instruction *cursor = generateRILInstruction(cg(), TR::InstOpCode::LARL, callNode, regEP, interfaceDataSymRef, interfaceDataSymbol->getStaticAddress());
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg());
+   cursor = generateRILInstruction(cg(), TR::InstOpCode::LARL, callNode, regRA, doneLabel, cursor);
+
+   TR::RegisterDependencyConditions *preDeps = new (trHeapMemory()) TR::RegisterDependencyConditions(dependencies->getPreConditions(), NULL,
+                                                                                                         dependencies->getAddCursorForPre(), 0, cg());
+   
+   TR::RegisterDependencyConditions *postDeps = new (trHeapMemory()) TR::RegisterDependencyConditions(dependencies, 0, numberOfAdditionalPostDeps, cg());
+   // Removing the pre dependencies from the postDeps
+   postDeps->setAddCursorForPre(0);
+   postDeps->setNumPreConditions(0, trMemory());
```

Comment:
Why is this needed?

Reply:
postDeps is a copy of existing `dependencies` to make room for new post dependencies I need for this function. Actually I can remove those extra lines of code as I have done in https://github.com/eclipse/openj9/compare/3282a01df297676d33999fca192d2613b5db1b09..d1f154c59f4e002f94b5e613001eb016a95a6927

Reply:
It is actually not needed, I wanted to just extract the post deps from the original dependencies and add room for extra deps I need in this function. There is a better way to achieve that and I have made change in https://github.com/eclipse/openj9/compare/3282a01df297676d33999fca192d2613b5db1b09..d1f154c59f4e002f94b5e613001eb016a95a6927